File: backend/src/main/java/com/ourride/model/entity/WalletTransaction.java
package com.ourride.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "wallet_transactions", indexes = {
    @Index(name = "idx_wallet_transactions_user_id", columnList = "userId"),
    @Index(name = "idx_wallet_transactions_type", columnList = "transactionType"),
    @Index(name = "idx_wallet_transactions_status", columnList = "status"),
    @Index(name = "idx_wallet_transactions_created_at", columnList = "createdAt")
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WalletTransaction {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "user_id", nullable = false)
    private Long userId;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "transaction_type", nullable = false, length = 20)
    private TransactionType transactionType;
    
    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal amount;
    
    @Column(name = "payment_method", length = 50)
    private String paymentMethod;
    
    @Column(name = "payment_method_details", length = 200)
    private String paymentMethodDetails;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private TransactionStatus status;
    
    @Column(name = "transaction_id", unique = true, length = 100)
    private String transactionId;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
    
    public enum TransactionType {
        TOP_UP, RIDE_PAYMENT, REFUND, WITHDRAWAL
    }
    
    public enum TransactionStatus {
        PENDING, COMPLETED, FAILED, CANCELLED
    }
}

File: mobile/client/lib/features/wallet/domain/entities/wallet_transaction.dart
import 'package:equatable/equatable.dart';

class WalletTransaction extends Equatable {
  final int id;
  final String transactionType; // TOP_UP, RIDE_PAYMENT, REFUND, WITHDRAWAL
  final double amount;
  final String? paymentMethod;
  final String? paymentMethodDetails;
  final String status; // PENDING, COMPLETED, FAILED, CANCELLED
  final String? transactionId;
  final String? description;
  final DateTime createdAt;

  const WalletTransaction({
    required this.id,
    required this.transactionType,
    required this.amount,
    this.paymentMethod,
    this.paymentMethodDetails,
    required this.status,
    this.transactionId,
    this.description,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [
        id,
        transactionType,
        amount,
        paymentMethod,
        paymentMethodDetails,
        status,
        transactionId,
        description,
        createdAt,
      ];
}

